subseas_flex_fcst/maproom.py

```diff
@@ -45,26 +48,75 @@
 
 APP.layout = layout.app_layout
 
-
-def read_cptdataset(y_transform=False):
-
-    fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+#function to open all datasets and then combine them
```

Comment:
You're right that we needn't open all of the files in order to get the data for a single (S, L) pair, and doing that is probably what's making it slow.

I encouraged Drew to write something like this `combine_cptdataset` function because, as you say, I have in mind a future in which all datasets will be available in a format like zarr that lets you quickly open a multi-file dataset as as a single multidimensional array. I wanted to build this maproom in a way that would make it easy to migrate to that future data environment. But in retrospect, that was bad advice, because it's so slow.

Keep a copy of `combine_cptdataset` somewhere, because we'll probably eventually want to write a zarrification script, and that will be the core of it. But for now rewrite `read_cptdataset` to open just one file instead of all of them.

Reply:
... and when it comes time to read from zarr instead of netcdf, it will be easy to reimplement `read_cptdataset` to do that.

Reply:
That might make obsolete a bunch of other problems if S and L are no more dimensions of the xarrays. Unless we'd want to try and keep clean 1-point such dims?

Reply:
Alright so I will work on redoing the code.. anywhere smart I might save the code I have already written for later?

Reply:
OR... is there time to implement the zarrification process to this project? Maybe not. I'd like this to work for seasonal forecasts by the trainings starting in mid-Ocotber and we're not quite there yet. And there are other things to attend (partners' DLs updates, onset python maprooms, prepare for training). That would also need to work with an updating framework since there will be new seasonal forecasts at partners' sites monthly. And so we'd need to train them on that as well. So probably a bit tight, especially given Aaron didn't suggest that in the first place.

Reply:
As for where to put the code. Maybe it makes sense to start another .py file, maybe predictions.py as there are some functions that are surfacing as needed and that would potentially be needed by other tools. There is combine and read cpt datasets. We need to make a function out of what computes the target start/end from S/L and we'll need more like this for other type of forecasts.

Reply:
Okay, I changed the code to open only a single ds at a time, pushing those changes soon to this PR for review. I have the old function saved in a temporary file - I can make a separate predictions.py file and move all the functions into that file if that is direction we are moving in with all of the functions for this app.